llvm/lib/CheerpUtils/IdenticalCodeFolding.cpp

```diff
@@ -232,9 +234,15 @@ bool IdenticalCodeFolding::equivalentInstruction(const llvm::Instruction* A, con
 		{
 			const AllocaInst* a = cast<AllocaInst>(A);
 			const AllocaInst* b = cast<AllocaInst>(B);
-			return CacheAndReturn(equivalentType(a->getAllocatedType(), b->getAllocatedType()) &&
+			if ((allocaPairs.count(a) && allocaPairs[a] != b) || (allocaPairs.count(b) && allocaPairs[b] != a))
```

Comment:
use allocaPairs.find(a) + comparison with end() instead of count()+array access.

You do the lookup only once this way